main.go

```diff
@@ -82,9 +107,70 @@ func (c *Cmd) Run() {
 	if err := c.cmd.Run(); err != nil {
 		c.exitError = err
 	}
-	c.stdout = string(outBuf.Bytes())
-	c.stderr = string(errBuf.Bytes())
-	c.executed = true
+	c.stdout.content = string(outBuf.Bytes())
+	c.stderr.content = string(errBuf.Bytes())
+	c.status = "executed"
+}
+
+// Start starts the command without waiting for it to complete
+func (c *Cmd) Start() {
+	if c.stdin != nil {
+		c.cmd.Stdin = c.stdin
+	}
+
+	if c.env != nil {
+		c.cmd.Env = c.env
+	} else {
+		c.cmd.Env = os.Environ()
+	}
+
+	stdoutPipe, err := c.cmd.StdoutPipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	stderrPipe, err := c.cmd.StderrPipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := c.cmd.Start(); err != nil {
+		c.exitError = err
+	}
+
+	go func() {
+		scanner := bufio.NewScanner(stdoutPipe)
+		for scanner.Scan() {
+			c.stdout.mu.Lock()
+			c.stdout.content += scanner.Text()
+			c.stdout.mu.Unlock()
```

Comment:
I'm curious to know what type of concurrency issues you ran into.

Reply:
I hope https://github.com/rendon/testcli/pull/8#discussion_r579804006 explains this well enough.